src/DynamoCore/Graph/Notes/NoteModel.cs

```diff
@@ -24,6 +26,25 @@ public string Text
             }
         }
 
+        private NodeModel pinnedNode;
+
+        /// <summary>
+        /// NodeModel which this Note is pinned to
+        /// When using the pin to node command  
+        /// note and node become entangled so that 
+        /// if you select and move one the other one 
+        /// moves as well. 
+        /// </summary>
+        public NodeModel PinnedNode
```

Comment:
hi @martinstacey  can you specify the reason to keep a reference on the Model level?

Reply:
This is needed as serialization both for saving the file and copying/pasting use model references.